### PR TITLE
20 merchant invoice show page update item status

### LINKED ISSUE #20: Merchant Invoice Show Page: Update Item Status
Merchant Invoice Show Page: Update Item Status

As a merchant
When I visit my merchant invoice show page
I see that each invoice item status is a select field
And I see that the invoice item's current status is selected
When I click this select field,
Then I can select a new status for the Item,
And next to the select field I see a button to "Update Item Status"
When I click this button
I am taken back to the merchant invoice show page
And I see that my Item's status has now been updated

### PR DESCRIPTION
**Pull request by rebeckahendricks**
*Create feature test for Update Item Status feature
*Create Update Item Status feature in Merchant Invoice Show page
*Fix test issue with dates for dashboard_spec
(Pair programmed with Thomas)

### CODE REVIEW

**Review by lcole37 — APPROVED** [review]

File: app/views/merchant/invoices/show.html.erb [hunk 1]
```diff
@@ -7,11 +7,17 @@
...
 <% @invoice_items.each do |invoice_item| %>
   <div id=<%= "#{invoice_item.item.id}" %>>
-    <p>Item Name: <%= invoice_item.item.name %>
+    <h3>Item Name: <%= invoice_item.item.name %></h3>
     <p>Item Quantity: <%= invoice_item.quantity %>
     <p>Item Unit Price: $<%= '%.2f'% ((invoice_item.unit_price.to_f) / 100) %>
-    <p>Item Status: <%= invoice_item.status %>
+    <h4>Item Status: </h4>
+      <div id="status_<%= "#{invoice_item.item.id}" %>">
+        <%= form_with url: merchant_invoice_path(@merchant, @invoice), method: :patch do |form| %>
```
Comment (thayes87) [inline app/views/merchant/invoices/show.html.erb:15]: This as fun creating a form_with drop down menu

File: spec/features/merchant/invoices/show_spec.rb [hunk 2]
```diff
@@ -42,6 +41,85 @@
         visit merchant_invoice_path(@merchant, invoice_1)
           expect(page).to have_content("Total Revenue: $21067.77")
       end
+
+      it 'I see that each invoice item status is a select field and I see that the invoice items current status is selected' do
```
Comment (thayes87) [inline spec/features/merchant/invoices/show_spec.rb:45]: When in doubt - Write a Test!

File: app/controllers/merchant/invoices_controller.rb [hunk 3]
```diff
@@ -6,7 +6,21 @@ def index
...
+    @invoice = Invoice.find(params[:id])
+    @invoice_items = InvoiceItem.for_merchant(params[:merchant_id])
+  end
+
+  def update
+    @merchant = Merchant.find(params[:merchant_id])
     @invoice = Invoice.find(params[:id])
     @invoice_items = InvoiceItem.for_merchant(params[:merchant_id])
+    @invoice_item = InvoiceItem.update(invoice_item_params)
+    render :show
```
Comment (thayes87) [inline app/controllers/merchant/invoices_controller.rb:19]: Classic Render

**Review by thayes87 — APPROVED** [review]
This was an enjoyable story to work on together!

**Pull request merged**